app/services/fees/create_pay_in_advance_service.rb

```diff
@@ -101,7 +102,7 @@ def boundaries
         charges_from_datetime: date_service.charges_from_datetime,
         charges_to_datetime: date_service.charges_to_datetime,
         charges_duration: date_service.charges_duration_in_days,
-        timestamp: Time.current
+        timestamp: Time.current # QUESTION: should we use billing_at ?
```

Comment:
I believe `Time.current` was used as a shortcut. Usually in boundaries, the timestamp is "when the thing was billed". It should have been billed at `billing_at` but the execution can be delayed for technical reason (sidekiq queue being late, job failed and is tried later...).

Or am I wrong and the timestamp should reflect "the time it was actually executed"?

Reply:
@julienbourdeau you are right, it should definitely be `billing_ag`